pkg/systemd/hw-detect.es6

```diff
@@ -55,8 +55,49 @@ function findPCI(udevdb, info) {
     }
 }
 
+// Parse and add to info.dmi.dimm = [{"speed":"11 MHz","locator":"A3", ...}]
+function processDMIDE(res, info) {
+  var dimms = [];
+  var data = res.split("\n\n");
+  for (var i = 1; i < data.length - 1; i++) {
+      if (data[i].length == 0) {
+        continue;
+      }
+      var dict = {};
+      var speed = data[i].substring(data[i].indexOf("Speed: "), data[i].indexOf("Manufacturer") - 1);
```

Comment:
This is too brittle. We should not hardcode the assumption that these two fields always follow each other, or are always present. You should rather follow the actual structure, like a regexp `^\s*([^:)+:\s*(.*)$` so that group(1) is the key name and group(2) the value. Of course you still need to treat the section headers/blocks.

As said above, this generic structure parsing should already happen in the library above, as it's not specific to the caller. In this file it should merely pick out the keys that it wants.